molnsutil/molnsutil.py

```diff
@@ -561,23 +565,71 @@ def map_and_aggregate(results, param_set_id, mapper, aggregator=None, cache_resu
 
     #return res
 
+############################################################################
 class DistributedEnsemble():
     """ A class to provide an API for execution of a distributed ensemble. """
 
-    def __init__(self, model_class=None, parameters=None, client=None, num_engines=None):
+    my_class_name = 'DistributedEnsemble'
+
+    #-----------------------------------------------------------------------------------
+    @classmethod
+    def delete(cls, name):
+        """ Static method to remove the state of a distributed comptutation from the system."""
+        # delete realization
+        try:
+            with open('.molnsutil/{1}-{0}'.format(name, cls.my_class_name)) as fd:
+                state = pickle.load(fd)
+                
+                if state['storage_mode'] is not None:
+                    if state['storage_mode'] == "Shared":
+                        ss = SharedStorage()
+                    elif state['storage_mode'] == "Persistent":
+                        ss = PersistentStorage()
+                    for param_set_id in state['result_list']:
+                        for filename in state['result_list'][param_set_id]:
+                            try:
+                                ss.delete(filename)
+                            except OSError as e:
+                                pass
+            # delete database file
+            os.remove('.molnsutil/{1}-{0}'.format(name, cls.my_class_name))
+        except Exception as e:
+            sys.stderr.write('delete(): {0}'.format(e))
+
+    #-----------------------------------------------------------------------------------
+    def __init__(self, name=None, model_class=None, parameters=None, client=None, num_engines=None, ignore_model_mismatch=False):
         """ Constructor """
-        self.my_class_name = 'DistributedEnsemble'
+        if not isinstance(name, str):
+            raise MolnsUtilException("name not specified")
+        self.name = name
+        if not inspect.isclass(model_class):
+            raise MolnsUtilException("model_class not a class")
         self.model_class = cloudpickle.dumps(model_class)
-        self.parameters = [parameters]
-        self.number_of_trajectories = 0
-        self.seed_base = self.generate_seed_base()
-        self.storage_mode = None
-        # A chunk list
-        self.result_list = {}
         # Set the Ipython.parallel client
         self.num_engines = num_engines
         self._update_client(client)
-
+        try:
```

Comment:
Make it more explicit that you are trying to reconstruct the ensemble from the 'name' here, the flow control by try, except can be a bit confusing.